lib/ecto/query/builder.ex

```diff
@@ -82,6 +82,17 @@ defmodule Ecto.Query.Builder do
     {{:{}, [], [:fragment, [], merge_fragments(pieces, frags)]}, params_acc}
   end
 
+  def escape({:unsafe_fragment, _, [{:^, _, [var]} = _expr]}, _type, params_acc, _vars, _env) do
+    expr = quote do
+      Ecto.Query.Builder.binary!(unquote(var))
+    end
+    {{:{}, [], [:fragment, [], [{:raw, expr}]]}, params_acc}
+  end
+
+  def escape({:unsafe_fragment, _, _}, _type, _params_acc, _vars, _env) do
+    error! "unsafe_fragment(...) expects only an interpolated string"
```

Comment:
We can probably do better on the error message. In particular, interpolated string is amibiguous. Something like:

error! """
unsafe_fragment(...) expects a single argument to be interpolated as ^argument

The argument is a string that will be sent to the database as is. For this reason,
use this feature with extreme care and instead prefer to use fragment as much 
as possible.
"""
```

Can you also make sure `fragment/1` raises a nice error message if someone is trying to interpolate something into it?

Reply:
Ok for the error message.

About fragment/1 : there's already a check in place which makes sure that the interpolated is only a keyword list (via keyword!/1) . Do you mean to rephrase the message? Or something else?

Reply:
> About fragment/1 : there's already a check in place which makes sure that the interpolated is only a keyword list (via keyword!/1) . Do you mean to rephrase the message? Or something else?

Good point. I think we can improve that check so that, if a binary is given, we mention unsafe fragment and the same kind of pitfalls and so on.